src/ILCompiler/src/Program.cs

```diff
@@ -281,7 +282,17 @@ private int Run(string[] args)
             // TODO: compiler switch for SIMD support?
             var simdVectorLength = (_isCppCodegen || _isWasmCodegen) ? SimdVectorLength.None : SimdVectorLength.Vector128Bit;
             var targetDetails = new TargetDetails(_targetArchitecture, _targetOS, TargetAbi.CoreRT, simdVectorLength);
-            var typeSystemContext = new CompilerTypeSystemContext(targetDetails, genericsMode);
+            CompilerTypeSystemContext typeSystemContext;
+            ReadyToRunCompilerContext r2rCompilerContext = null;
```

Comment:
I think we can get away with just the `CompilerTypeSystemContext typeSystemContext;` variable now.
I would just do

```csharp
CompilerTypeSystemContext typeSystemContext = _isReadyToRunCodeGen ? new ReadyToRunCompilerContext(targetDetails, genericsMode) : new CompilerTypeSystemContext(targetDetails, genericsMode);
```

to avoid the ceremony. The context in the ReadyToRunCodegenCompilationBuilder ctor can stay typed as CompilerTypeSystemContext.

Reply:
Fixed in 10th commit :-).